src/main/scala/analysis/GlobalRegionAnalysis.scala

```diff
@@ -0,0 +1,246 @@
+package analysis
+
+import analysis.solvers.SimpleWorklistFixpointSolver
+import ir.*
+
+import scala.collection.mutable
+
+trait GlobalRegionAnalysis(val program: Program,
+                           val domain: Set[CFGPosition],
+                           val constantProp: Map[CFGPosition, Map[Variable, FlatElement[BitVecLiteral]]],
+                           val reachingDefs: Map[CFGPosition, (Map[Variable, Set[Assign]], Map[Variable, Set[Assign]])],
+                           val mmm: MemoryModelMap,
+                           val globalOffsets: Map[BigInt, BigInt],
+                           val vsaResult: Option[Map[CFGPosition, LiftedElement[Map[Variable | MemoryRegion, Set[Value]]]]]) {
+
+  var dataCount: Int = 0
+  private def nextDataCount() = {
+    dataCount += 1
+    s"data_$dataCount"
+  }
+
+  val regionLattice: PowersetLattice[DataRegion] = PowersetLattice()
+
+  val lattice: MapLattice[CFGPosition, Set[DataRegion], PowersetLattice[DataRegion]] = MapLattice(regionLattice)
+
+  val first: Set[CFGPosition] = Set.empty + program.mainProcedure
+
+  private val stackPointer = Register("R31", 64)
+  private val linkRegister = Register("R30", 64)
+  private val framePointer = Register("R29", 64)
+  private val mallocVariable = Register("R0", 64)
+
+  private val dataMap: mutable.HashMap[BigInt, DataRegion] = mutable.HashMap()
+
+  private def dataPoolMaster(offset: BigInt, size: BigInt): Option[DataRegion] = {
+    assert(size >= 0)
+    if (dataMap.contains(offset)) {
+      if (dataMap(offset).size < (size.toDouble / 8).ceil.toInt) {
+        dataMap(offset) = DataRegion(dataMap(offset).regionIdentifier, offset, (size.toDouble / 8).ceil.toInt)
+        Some(dataMap(offset))
+      } else {
+        Some(dataMap(offset))
+      }
+    } else {
+      dataMap(offset) = DataRegion(nextDataCount(), offset, (size.toDouble / 8).ceil.toInt)
+      Some(dataMap(offset))
+    }
+  }
+
+  def getDataMap: mutable.HashMap[BigInt, DataRegion] = dataMap
+
+  def resolveGlobalOffsetSecondLast(address: BigInt): BigInt = {
+    var tableAddress = address
+    // addresses may be layered as in jumptable2 example for which recursive search is required
+    var exitLoop = false
+    while (globalOffsets.contains(tableAddress) && globalOffsets.contains(globalOffsets(tableAddress)) && !exitLoop) {
+      val newAddress = globalOffsets.getOrElse(tableAddress, tableAddress)
+      if (newAddress == tableAddress) {
+        exitLoop = true
+      } else {
+        tableAddress = newAddress
+      }
+    }
+    tableAddress
+  }
```

Comment:
I think this is the cause of the issue with pointers-to-pointers being given the same region as the pointers they point to. This is not correct - you can't just replace a pointer-to-a-pointer with said pointer like this.